src/merkleDistributor/MerkleDistributor.ts

```diff
@@ -0,0 +1,30 @@
+import { ethers } from "ethers";
+import { DistributionRecipientsWithProofs, DistributionRecipients, MerkleTree } from "./model";
+
+export class MerkleDistributor {
+  static createMerkleDistributionProofs = (recipients: DistributionRecipients, windowIndex: number) => {
+    const recipientLeafs = Object.keys(recipients).map((addr: string) =>
+      MerkleDistributor.createLeaf(addr, recipients[addr].amount, recipients[addr].accountIndex)
+    );
+    const merkleTree = new MerkleTree(recipientLeafs);
+    const recipientsWithProofs = Object.keys(recipients).reduce((acc, recipientAddress, index) => {
+      return {
+        ...acc,
+        [recipientAddress]: {
+          ...recipients[recipientAddress],
+          proofs: merkleTree.getHexProof(recipientLeafs[index]),
+          windowIndex,
+        },
+      };
+    }, {} as DistributionRecipientsWithProofs);
+    return { recipientsWithProofs, merkleRoot: merkleTree.getHexRoot() };
+  };
+
+  // keccak256(abi.encode(account, amount))
+  static createLeaf = (account: string, amount: string, accountIndex: number) => {
```

Comment:
With the merkle distributor class from @across-protocol/contracts-v2, you can just pass a function like this into the class and then the class will take care of creating leaves for you.

Reply:
👍